src/window/mod.rs

```diff
@@ -235,9 +237,12 @@ pub fn main_loop(
     event_loop: EventLoop<UserEvent>,
 ) -> Result<(), EventLoopError> {
     let cmd_line_settings = SETTINGS.get::<CmdLineSettings>();
+    let render_thread_done = Arc::<AtomicBool>::new(AtomicBool::new(false));
+    let render_thread_done2 = render_thread_done.clone();
     let (txtemp, rx) = channel::<Event<UserEvent>>();
     let mut tx = Some(txtemp);
-    let mut render_thread_handle = Some(thread::spawn(move || {
+
+    let render_thread_handle = thread::spawn(move || {
```

Comment:
Rather than carrying a temporary variable here with a clumsy 2 suffix, I'd wrap the clojure which uses it in a block and shadow the original variable name like this:

```suggestion
    let render_thread_handle = thread::spawn({
        let render_thread_done = render_thread_done.clone();
        move || {
        ...
    }});
```

Then you don't need the second variable.

Reply:
Ah, I will try that. I basically copied the style in that vsync thread for this, but I don't like the 2 suffix much either.

Reply:
I fixed this one as well. I used `Arc::clone()` instead of `.clone()` since it's more idiomatic and doesn't trigger clippy's `clone_on_ref_ptr` lint.